parquet-variant/src/test/java/org/apache/parquet/variant/TestVariantObject.java

```diff
@@ -250,22 +212,61 @@ public void testMixedObject() {
     Variant value = new Variant(
         ByteBuffer.wrap(constructObject(keys, fields, true)),
         constructMetadata(true, ImmutableList.of("a", "b", "c")));
-    testVariant(value, v -> {
-      checkType(v, VariantUtil.OBJECT, Variant.Type.OBJECT);
+    VariantTestUtil.testVariant(value, v -> {
+      VariantTestUtil.checkType(v, VariantUtil.OBJECT, Variant.Type.OBJECT);
       Assert.assertEquals(3, v.numObjectElements());
-      checkType(v.getFieldByKey("a"), VariantUtil.PRIMITIVE, Variant.Type.INT);
+      VariantTestUtil.checkType(v.getFieldByKey("a"), VariantUtil.PRIMITIVE, Variant.Type.INT);
       Assert.assertEquals(1234567890, v.getFieldByKey("a").getInt());
-      checkType(v.getFieldByKey("b"), VariantUtil.PRIMITIVE, Variant.Type.BOOLEAN);
+      VariantTestUtil.checkType(v.getFieldByKey("b"), VariantUtil.PRIMITIVE, Variant.Type.BOOLEAN);
       Assert.assertTrue(v.getFieldByKey("b").getBoolean());
-      checkType(v.getFieldByKey("c"), VariantUtil.OBJECT, Variant.Type.OBJECT);
+      VariantTestUtil.checkType(v.getFieldByKey("c"), VariantUtil.OBJECT, Variant.Type.OBJECT);
 
       Variant nestedV = v.getFieldByKey("c");
       Assert.assertEquals(2, nestedV.numObjectElements());
-      checkType(nestedV.getFieldByKey("a"), VariantUtil.PRIMITIVE, Variant.Type.DATE);
+      VariantTestUtil.checkType(nestedV.getFieldByKey("a"), VariantUtil.PRIMITIVE, Variant.Type.DATE);
       Assert.assertEquals(
           LocalDate.parse("2025-04-17"),
           LocalDate.ofEpochDay(nestedV.getFieldByKey("a").getInt()));
-      checkType(nestedV.getFieldByKey("c"), VariantUtil.PRIMITIVE, Variant.Type.NULL);
+      VariantTestUtil.checkType(nestedV.getFieldByKey("c"), VariantUtil.PRIMITIVE, Variant.Type.NULL);
+    });
+  }
+
+  @Test
+  public void testMixedObjectBuilder() {
+    VariantBuilder b = new VariantBuilder(true);
+    VariantObjectBuilder objBuilder = b.startObject();
+    b.appendObjectKey(objBuilder, "outer 1");
+    b.appendBoolean(true);
+    b.appendObjectKey(objBuilder, "outer 2");
+    b.appendLong(1234567890);
+    b.appendObjectKey(objBuilder, "outer 3");
+    {
+      // build a nested obj
+      VariantObjectBuilder nestedBuilder = b.startObject();
+      b.appendObjectKey(nestedBuilder, "nested 1");
+      {
+        // build a nested empty obj
+        VariantObjectBuilder emptyBuilder = b.startObject();
+        b.endObject(emptyBuilder);
+      }
+      b.appendObjectKey(nestedBuilder, "nested 2");
+      b.appendString("variant");
+      b.endObject(nestedBuilder);
+    }
+    b.endObject(objBuilder);
+
+    VariantTestUtil.testVariant(b.build(), v -> {
+      VariantTestUtil.checkType(v, VariantUtil.OBJECT, Variant.Type.OBJECT);
+      Assert.assertEquals(3, v.numObjectElements());
+      Assert.assertTrue(v.getFieldByKey("outer 1").getBoolean());
+      Assert.assertEquals(1234567890, v.getFieldByKey("outer 2").getLong());
```

Comment:
I think this assertion needs to verify that the field was encoded as a long because `appendLong` was used. Can you add assertions for the types as well?